src/schema/resolvers.js

```diff
@@ -749,11 +762,28 @@ module.exports = {
 
   Topic: {
     oracles: async ({ address }, data, { db: { Oracles } }) => Oracles.find({ topicAddress: address }),
-    transactions: async ({ address }, data, { db: { Transactions } }) => Transactions.find({ topicAddress: address }),
+    transactions: async ({ address }, data, { db: { Transactions } }) => {
+      const types = [{ type: 'WITHDRAWESCROW' }, { type: 'WITHDRAW' }];
+      return Transactions.find({ topicAddress: address, $or: types });
+    },
   },
 
   Oracle: {
-    transactions: async ({ address }, data, { db: { Transactions } }) => Transactions.find({ oracleAddress: address }),
+    transactions: async (oracle, data, { db: { Transactions } }) => {
+      const { isBetting, isVoting, isResultSetting, isFinalizing } = getPhase(oracle);
+
+      let types = [];
+      if (isBetting) {
```

Comment:
lets use a `switch` statement (its faster computationally) and throw on an invalid phase

Reply:
totally cool with this, the only thing is, how do we do that with boolean values/an object that looks like:
```js
phase = {
  betting: true,
  voting: false,
  resultSetting: false,
  finalizing: false
}
```
we'd have to make the `getPhase` differently. Like this:
```js
const getPhase = ({ token, status }) => {
  if (token === 'QTUM' && status === 'VOTING') return 'betting';
  if (token === 'BOT' && status === 'VOTING') return 'voting';
  if (token === 'QTUM' && ['WAITRESULT', 'OPENRESULTSET'].includes(status)) return 'resultSetting';
  if (token === 'BOT' && status === 'WAITRESULT') return 'finalizing';
  throw Error('invalid phase');
};
```
and then in the resolver be like
```js
const phase = getPhase(oracle);
switch (phase) {
  case 'betting':
    types = [{ type: 'BET' }, { type: 'CREATEEVENT' }, { type: 'APPROVECREATEEVENT' }];
  case 'voting':
    types = [{ type: 'VOTE' }, { type: 'APPROVEVOTE' }];
  case 'isResultSetting':
    types = [{ type: 'SETRESULT' }, { type: 'APPROVESETRESULT' }]
    break;
  case 'finalizing':
    types = [{ type: 'FINALIZERESULT' }];
    break;
}
```
totally chill doing it that way^ just wanted to show the full diference

Reply:
unless you were thinking of a different way? 🤔

Reply:
ya, that's what i was thinking. let's return `getPhase` as a type and use a switch with that.